packages/eventual-send/src/index.test-d.ts

```diff
@@ -46,3 +46,15 @@ const foo2 = async (a: FarRef<{ bar(): string; baz: number }>) => {
   // @ts-expect-error - calling directly is valid but not yet in the typedef
   a.bar;
 };
+
+// when
```

Comment:
If we're adding `when` type tests, do you mind also adding a test for the "normal" use case with a `onfulfilled` callback?